optimizations-server/src/sft/common.py

```diff
@@ -34,6 +36,60 @@ class ParsedInferenceExample(TypedDict):
 T = t.TypeVar("T")
 
 
+class ValidationError(Exception):
+    pass
+
+
+class BaseSFTJob(BaseModel):
+    @abstractmethod
+    async def poll(self) -> "BaseSFTJob": ...
+
+    @abstractmethod
+    def status(self) -> t.Any: ...
+
+
+def render_message(content: t.Dict[str, Any], role: str, env: Environment) -> str:
+    assert role in ["user", "assistant"], f"Invalid role: {role}"
+
+    if content["type"] != "text":
+        raise ValueError(f"Content block must be of type text: {content}")
+
+    content = content["value"]
+
+    if isinstance(content, str):
+        return content
+    else:
+        return env.render_template(role, **content)
+
+
+def try_template_system(
+    sample: t.Dict[str, t.Any], env: Environment
+) -> t.Optional[t.Dict[str, str]]:
+    system = sample["input"].get("system")
+    if system is not None:
+        # TODO - add a 'has_template' to the minijinja python bindings
```

Comment:
@Aaron1011 we don't maintain the minijinja python bindings so it might have to be an external PR (annoying). Is there another way to do it?

Reply:
Not that I can tell, unfortunately - it looks like the only way to detect a missing template is to catch the error